packages/datadog-instrumentations/src/playwright.js

```diff
@@ -0,0 +1,236 @@
+    remainingTestsByFile = getTestsBySuiteFromTestsById(this._testById)
+    return run.apply(this, arguments)
+  }
+}
+
+function dispatcherHook (dispatcherExport) {
+  shimmer.wrap(dispatcherExport.Dispatcher.prototype, 'run', dispatcherRunWrapper)
+  shimmer.wrap(dispatcherExport.Dispatcher.prototype, '_createWorker', createWorker => function () {
+    const dispatcher = this
+    const worker = createWorker.apply(this, arguments)
+
+    worker.process.on('message', ({ method, params }) => {
+      if (method === 'testBegin') {
+        const { test } = dispatcher._testById.get(params.testId)
+        testBeginHandler(test)
+      } else if (method === 'testEnd') {
+        const { test } = dispatcher._testById.get(params.testId)
+
+        const { results } = test
+        const testResult = results[results.length - 1]
+
+        testEndHandler(test, STATUS_TO_TEST_STATUS[testResult.status], testResult.error)
+      }
+    })
+
+    return worker
+  })
+  return dispatcherExport
+}
+
+function dispatcherHookNew (dispatcherExport) {
+  shimmer.wrap(dispatcherExport.Dispatcher.prototype, 'run', dispatcherRunWrapper)
+  shimmer.wrap(dispatcherExport.Dispatcher.prototype, '_createWorker', createWorker => function () {
+    const dispatcher = this
+    const worker = createWorker.apply(this, arguments)
+
+    worker.on('testBegin', ({ testId }) => {
+      const { test } = dispatcher._testById.get(testId)
+      testBeginHandler(test)
+    })
+    worker.on('testEnd', ({ testId, status, errors }) => {
+      const { test } = dispatcher._testById.get(testId)
+
+      testEndHandler(test, STATUS_TO_TEST_STATUS[status], errors[0])
+    })
+
+    return worker
+  })
+  return dispatcherExport
+}
+
+function runnerHook (runnerExport) {
+  shimmer.wrap(runnerExport.Runner.prototype, 'runAllTests', runAllTests => async function () {
+    const testSessionAsyncResource = new AsyncResource('bound-anonymous-fn')
+    const { version: frameworkVersion } = getPlaywrightConfig(this)
+
+    const rootDir = getRootDir(this)
+
+    const processArgv = process.argv.slice(2).join(' ')
+    const command = `playwright ${processArgv}`
+    testSessionAsyncResource.runInAsyncScope(() => {
+      testSessionStartCh.publish({ command, frameworkVersion, rootDir })
+    })
+
+    const res = await runAllTests.apply(this, arguments)
+    const sessionStatus = STATUS_TO_TEST_STATUS[res.status]
+
+    let onDone
+
+    const flushWait = new Promise(resolve => {
+      onDone = resolve
+    })
+
+    testSessionAsyncResource.runInAsyncScope(() => {
+      testSessionFinishCh.publish({ status: sessionStatus, onDone })
+    })
+    await flushWait
+
+    startedSuites = []
+    remainingTestsByFile = {}
+
+    return res
+  })
+
+  return runnerExport
+}
+
+addHook({
+  name: '@playwright/test',
+  file: 'lib/runner.js',
+  versions: ['>=1.18.0']
+}, runnerHook)
+
+addHook({
+  name: '@playwright/test',
+  file: 'lib/dispatcher.js',
+  versions: ['1.18.0 - 1.29.2']
```

Comment:
If they release a patch for `1.29.x` we are going to ignore it... what do you think about this version range?

```suggestion
  versions: ['>=1.18.0  <1.30.0']
```

Reply:
good catch 👍